api/transport/router.go

```diff
@@ -25,33 +25,25 @@ import "context"
 // TODO: Until golang/mock#4 is fixed, imports in the generated code have to
 // be fixed by hand. They use vendor/* import paths rather than direct.
 
-//go:generate mockgen -destination=transporttest/register.go -package=transporttest go.uber.org/yarpc/api/transport Registry,Registrar
+//go:generate mockgen -destination=transporttest/router.go -package=transporttest go.uber.org/yarpc/api/transport Router,RouteTable
 
-// ServiceProcedure represents a service and procedure registered against a
-// Registry.
-type ServiceProcedure struct {
-	Service   string
-	Procedure string
-}
-
-// Registrant specifies a single handler registered against the registry.
-type Registrant struct {
-	// Service name or empty to use the default service name.
-	Service string
+// Procedure specifies a single handler registered in the RouteTable.
+type Procedure struct {
+	// ServiceName or empty to use the default service name.
+	ServiceName string
 
-	// Name of the procedure.
-	Procedure string
+	// ProcedureName of the procedure.
+	ProcedureName string
```

Comment:
Reduce stuttering

    procedure := Procedure{ProcedureName: ...}
    fmt.Println(procedure.ProcedureName)

Rename this attribute to Name and optionally make it the first attribute in the list.